sui/src/sui_node.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::path::Path;
+
+use crate::{
+    api::{RpcGatewayServer, SignedTransaction, TransactionBytes},
+    rpc_gateway::responses::{GetObjectInfoResponse, ObjectResponse, SuiTypeTag},
+};
+use anyhow::anyhow;
+use async_trait::async_trait;
+use jsonrpsee::core::RpcResult;
+use move_core_types::identifier::Identifier;
+
+use sui_core::gateway_state::{
+    gateway_responses::{TransactionEffectsResponse, TransactionResponse},
+    GatewayTxSeqNumber,
+};
+use sui_core::sui_json::SuiJsonValue;
+use sui_types::{
+    base_types::{ObjectID, SuiAddress, TransactionDigest},
+    json_schema::Base64,
+    object::ObjectRead,
+};
+
+pub struct SuiNode {}
+
+impl SuiNode {
+    pub fn new(_config_path: &Path) -> anyhow::Result<Self> {
+        Ok(Self {})
+    }
+}
+
+#[async_trait]
+impl RpcGatewayServer for SuiNode {
```

Comment:
Yeah I wonder if we couldn't, as a pre-step, split out the read and write apis into separate traits.

Reply:
That's my inclination, but I wanted to get feedback first.

Reply:
We can split the apis into separate traits, and instantiate them as different modules in the JSON-RPC server.

However it will be good if we can kick this can down the road if possible? (trying to prevent a rebase hell :( ), I am currently working on improving the API's response types, I am creating a set of new types for the Gateway instead of passing internal types (`Object`, `MoveObject`, `MoveType` etc) to downstream, as they doesn't work well with serde_json. Decoupling the object type at the GatewayAPI boundary gives us the flexibility to make the response more JSON friendly.

Reply:
Sure, I can wait until your PR is merged. I assume you need me to wait on this whole PR, not splitting RpcGateway?

Reply:
Thanks!!
This PR is fine :) The changes in RpcGateway is quite straightforward.